frontend/src/pages/pipelines/global/experiments/executions/ExecutionsTableRowStatusIcon.tsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import { Icon, Tooltip } from '@patternfly/react-core';
+import {
+  CheckCircleIcon,
+  ExclamationCircleIcon,
+  OutlinedWindowRestoreIcon,
+  QuestionCircleIcon,
+  TimesCircleIcon,
+} from '@patternfly/react-icons';
+import { Execution } from '~/third_party/mlmd';
+
+type ExecutionsTableRowStatusIconProps = {
+  status: Execution.State;
+};
+
+const ExecutionsTableRowStatusIcon: React.FC<ExecutionsTableRowStatusIconProps> = ({ status }) => {
+  let tooltip;
+  let icon;
+  switch (status) {
+    case Execution.State.COMPLETE:
+      icon = (
+        <Icon status="success">
+          <CheckCircleIcon />
+        </Icon>
+      );
+      tooltip = 'Complete';
+      break;
+    case Execution.State.CACHED:
+      icon = (
+        <Icon status="custom">
+          <OutlinedWindowRestoreIcon />
+        </Icon>
+      );
+      tooltip = 'Cached';
+      break;
+    case Execution.State.CANCELED:
+      icon = (
+        <Icon>
+          <TimesCircleIcon />
+        </Icon>
+      );
+      tooltip = 'Canceled';
+      break;
+    case Execution.State.FAILED:
+      icon = (
+        <Icon status="danger">
+          <ExclamationCircleIcon />
+        </Icon>
+      );
+      tooltip = 'Failed';
+      break;
+    case Execution.State.RUNNING:
+      icon = <Icon isInProgress />;
+      tooltip = 'Running';
+      break;
+    // TODO: change the icon here
```

Comment:
is this todo still valid

Reply:
Yes, I need to double-check with UX, thanks for the reminder. I can leave this comment here for now and update it later when I get the feedback from the UX.